### DISCUSSION TITLE
Alert Levels

### DISCUSSION THREAD

**Discussion by Xansta** [Thorium-Sim/thorium-nova#43] [General]
Split off from #41 (read there for context)

Three sounds reasonable to me.

I'm not saying we have to restrict actions based on alert levels ala Starship Horizons, but we should make it something that can be configured either based on the scenario or perhaps by ship type. Maybe the Federation ships require the alert level be set to red before weapons can be fired, but the Klingon ships have no such restriction since they are always ready to shoot any dishonorable entity nearby.

I don't have direct SBS experience with "theme controls visual aspects" since my primary experience is with EE and Artemis where there is but one theme. I agree we should design themes carefully. In the design, we should consider that a yellow alert or red alert that has no impact to the actions you can take should not pour buckets of yellow and red pixels all over your screen.

**Comment by LordAuric** [comment]
Just make Alert levels configurable like Horizons, where it's based on the vessel class definition. Allows for a lot of flexibility, and doesn't lock you in.

**Reply by alexanderson1993** [reply]
Tell us more about that. How can you configure them in Horizons?

**Reply by LordAuric** [reply]
Horizons has Vessel Class definitions, that outline each type of ship in the universe. On this definition you can define the Alert statues, which correlate to operational conditions. On each status you can define it's Name, Color, Priority (Higher is worse... aka Red Alert), and then a Collection of Tasks that are allowed. Examples include weapon loading, weapon firing, defensive component use (Shields), and crew state (Standard, High Alert and Battle Stations). The AI crew react differently based on which is active. As mentioned above, a capitol ship will likely use these states, but a Fighter craft would not.

**Reply by alexanderson1993** [reply]
> a capitol ship will likely use these states, but a Fighter craft would not

I legit never considered this, but it makes perfect sense.

**Reply by aBlueShadow** [reply]
Some other Idea: Instead of letting alert levels restrict actions, actions/events could trigger alert levels.
Like:
raise shields -> sets yellow alert
fire torpedo/hull gets damaged -> sets red alert

and vice versa:
set green alert -> "Warning, this will lower your shields. Are you sure? Y/N" or just a message "Cannot set to green while shields are active". So settings would restrict alert levels instead of the other way around-

**Reply by aBlueShadow** [reply]
Alternatively: if settings are limited because of alert settings, there should be at least some clear indication for it IMO.
